validation/parser/pipeline/parser_pipeline_funcs.py

```diff
@@ -0,0 +1,32 @@
+from parser.pair import Pair
```

Comment:
Given that these functions are each only used in one file, it would make more sense to move them into the file where they're used to reduce the cognitive load of switching between modules.

Reply:
I originally had it that way (previous commit), but I thought it would be a cool idea to have a module of functions that can be used by a ParserPipeline. So for now we only use some of them in one file, for instance remove_empty. However, in the future as more flags are added, these functions can be mixed and matched with other parsers. Basically any function in pipeline_functions can take in an iterable of pairs and return an iterable of pairs, and they can be used throughout different modules